blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
yeah, the XAResourceManager::set_complete(XAWrapper* resource)
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManager.cxx#L328

it delete the ra and the branch->get_name() could get the wrong memory reference.